lib/takeWhile.js

```diff
@@ -0,0 +1,13 @@
+/*
+ * `takeWhile` take a slice of an array, elements will be taken 
+ * from the begining of the array until selector returns false
+ * 
+ * @params {Array} array from which values will be iterated
+ * @params {predicate} delegate function to determine when to stop iterating
```

Comment:
I'm sorry. I wasn't clear. Function is the right type. I was just suggesting delegate->predicate